cmd/discover-pd/main.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"flag"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/pingcap/tidb-operator/pkg/discovery"
+	"github.com/pingcap/tidb-operator/pkg/discovery/server"
+	"github.com/pingcap/tidb-operator/pkg/version"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apiserver/pkg/util/logs"
+)
+
+var (
+	printVersion bool
+	port         int
+	scheme       string
+	replicas     int
+)
+
+func init() {
+	flag.BoolVar(&printVersion, "V", false, "Show version and quit")
+	flag.BoolVar(&printVersion, "version", false, "Show version and quit")
+	flag.StringVar(&scheme, "scheme", "http", "protocol scheme")
+	flag.IntVar(&port, "port", 10261, "The port that the tidb discovery's http service runs on (default 10261)")
+	flag.IntVar(&replicas, "replicas", 3, "The number of PD replicas to expect")
+	flag.Parse()
+}
+
+func main() {
+	if printVersion {
+		version.PrintVersionInfo()
+		os.Exit(0)
+	}
+	version.LogVersionInfo()
+	RunDiscoveryService(port)
+}
+
+// RunDiscoveryService blocks forever
+// Errors in this function are fatal
+func RunDiscoveryService(port int) {
+	logs.InitLogs()
+	defer logs.FlushLogs()
+
+	td := discovery.NewTiDBDiscoveryImmediate(getCluster{})
+	go wait.Forever(func() {
+		server.StartServer(td, port)
+	}, 5*time.Second)
+	glog.Fatal(http.ListenAndServe(":6060", nil))
```

Comment:
I guess this is for pprof, if so you need to import `net/http/pprof` package.

Reply:
hmm, I copied this from the original code. I think it is preferable to just call StartServer and not run pprof.

Reply:
okay, I looked into pprof more, it looks like it doesn't create much overhead so we can run it in a goroutine. I don't understand though why we need `wait.Forever`.

Reply:
I am adding pprof in the main as a goroutine and getting rid of wait.Forever.